2.7.0/final.md

```diff
@@ -0,0 +1,1034 @@
+  * Better error when kernel fails to compile ([\#144649](https://github.com/pytorch/pytorch/pull/144649))
+  * Fix large prod and sum reductions ([\#148975](https://github.com/pytorch/pytorch/pull/148975))
+  * Adding support to MPSInductor for operators: `gamma`, `zeta`, `sinc`, `spherical_bessel_j0`, `entr` ([\#145341](https://github.com/pytorch/pytorch/pull/145341), [\#146465](https://github.com/pytorch/pytorch/pull/146465), [\#146539](https://github.com/pytorch/pytorch/pull/146539), [\#147650](https://github.com/pytorch/pytorch/pull/147650), [\#148128](https://github.com/pytorch/pytorch/pull/148128))
+
+## ROCm
+- Fix TunableOp UTs: Rotating Buffer (#143172)
+- Enable *_load_dwordx4 ISA for BFloat16 and Half. ([#141397](https://github.com/pytorch/pytorch/pull/141397))
+- Fix condition for small tensor tuning ([#144087](https://github.com/pytorch/pytorch/pull/144087))
+
+## XPU
+- Enable FP64 GEMM ([#140677](https://github.com/pytorch/pytorch/pull/140677))
+- Enable Sparse CSR support ([#144722](https://github.com/pytorch/pytorch/pull/144722))
+- Improve XPU Stream implemenation([#141123](https://github.com/pytorch/pytorch/pull/141123),[#141119](https://github.com/pytorch/pytorch/pull/141119),[#142347](https://github.com/pytorch/pytorch/pull/142347))
+- Enable XPU for Inductor MM Triton Kernel Benchmark ([#148237](https://github.com/pytorch/pytorch/pull/148237))
+- Align XPU `convolution_backward` output layout between fake tensor and real output tensor ([#146880](https://github.com/pytorch/pytorch/pull/146880))
+- Improve error handling and reporting in CMake files ([#149353](https://github.com/pytorch/pytorch/pull/149353))
+- Refine `torch.xpu.get_device_properties` API error message ([#144379](https://github.com/pytorch/pytorch/pull/144379))
+- Enable `nested_layer_norm` support for XPU ([#148593](https://github.com/pytorch/pytorch/pull/148593))
+- Generalize `is_big_gpu()` check in Inductor ([#143491](https://github.com/pytorch/pytorch/pull/143491))
+- Allow XPU device in sparse compressed tensor factory functions (#147306)
+
+## Profiler
+- Add optional flag to profiler to toggle external correlations (#143314)
+- Add delimeter in memory vizualizer to show where allocation addr begins (#147461)
+- Add last entry to truncated values in Kineto args (#148576)
+- Add profiler activity for HPU devices (#148182)
+- Add HPU availabilities to profiler (#149115)
+
+## torch.compile
+#### Dynamo
+- Better tracing support for user-defined `dict` subclasses (#143548)
+- Improved graph break messages for some common graph break sites (#146525)
+- Improved tracing of exceptions (#146492)
+- Remove a number of builtin and third-party modules from `trace_rules.py` skipfiles (#145856)
+- Remove some specialized variables for specific third-party classes (e.g. `transformers` `ModelOutput`) (#143567)
+- Compiled Autograd dropped annotation requirements for custom autograd functions (#146229, #146720)
```

Comment:
cc @zou3519 @xmfan @williamwen42 this was previously categorized as a deprecation but it does not appear to be one. Moved it to improvements instead, pls lmk if this is incorrect.